io/io/src/TStreamerInfoReadBuffer.cxx

```diff
@@ -609,6 +609,7 @@ Int_t TStreamerInfo::ReadBufferConv(TBuffer &b, const T &arr,  const TCompInfo *
          DOLOOP {
             UInt_t u;
             b >> u;
+            u |= kNotDeleted;  // by definition de-serialized object are not yet deleted.
```

Comment:
I cannot see any use of `u` except for `if ((u & kIsReferenced) != 0)` in the next line. Why set `kNotDeleted`?

Reply:
(And indeed, is it intentional that here we don't send `obj->fBits`?

Reply:
> I  cannot see any use of u
It is used inside the switch statement at line 630, isn't it?

> (And indeed, is it intentional that here we don't send `obj->fBits`?

It is the bizarre case where `fBits` has somehow been change from `UInt_t` to some other type (and yes, I am not how/why this would happen, the code exist I just tweaked it :) ).